trin-storage/src/versioned/mod.rs

```diff
@@ -0,0 +1,33 @@
+pub mod memory;
+pub mod sql;
+pub mod store;
+mod utils;
+
+use rusqlite::types::{FromSql, FromSqlError, ValueRef};
+use strum::{AsRefStr, Display, EnumIter, EnumString};
+
+pub use store::VersionedContentStore;
+pub use utils::create_store;
+
+#[derive(Copy, Clone, Debug, Display, Eq, PartialEq, EnumString, AsRefStr, EnumIter)]
+#[strum(serialize_all = "snake_case")]
+pub enum ContentType {
+    Beacon,
+    History,
+    State,
+}
+
+#[derive(Copy, Clone, Debug, Display, Eq, PartialEq, EnumString, AsRefStr, EnumIter)]
+#[strum(serialize_all = "snake_case")]
+pub enum StoreVersion {
+    InMemory,
+    LegacyContentData,
```

Comment:
`Legacy*` is also placeholder for now. We should have implementation of `VersionedContentStore` that uses current logic. I'm fine with removing them until we have implementations in place (not any time soon).